### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by brettsimons** [HiFi-LoFi/FFTConvolver#3]
Hello, I was hoping to use this library for a school project but I can't figure out a few things. I am providing it with the raw wave audio data in float format (not manipulated in any way so there are multiple samples per float) as well as the IR from a SOFA file but, unfortunately, it seems to be outputting jiberish. Would you be able to help me understand what inputs your methods are expecting? I have a feeling that I am either providing the wrong block size (not sure what this does other than affect the processing speed) or you are expecting only 1 audio sample per float.

**Comment by HiFi-LoFi** [comment]
Hi,

what do you mean by "there are multiple samples per float"? Usually, an impulse response (e.g. loaded from some .wav file) consists of many (!) audio samples, and one audio sample is typically represented by a float. So, yes, of course, the convolver expects exactly one audio sample per float. I have no idea how there could be more than one audio sample per float (except for compressed signals or stuff like this).

The convolution algorithm expects both the impulse response as well as the to-be-convolved audio chunks as raw audio buffers (32-bit float). Maybe you have more channels, then you typically need one convolver object per channel (maybe you mean this by "multiple samples per float").

Ciao, Uli

**Comment by brettsimons** [comment]
Hey Uli, when I say multiple samples per float, I'm referencing the raw audio buffer. When reading in a wave file with 16-bit audio, there will be two 16-bit samples in every 32-bit object. I finally figured out the problem and it was that I needed to split the 16-bit samples so that each sample was its own entry in the array. Next, I had to divide each sample by SHRT_MAX to get float values between -1 and 1. After doing that, I could use this library. Then I could take the output values and reverse the process to pack it back into a wave file. Here is my program for reference: https://github.com/DSVSCA/DSVSCA/blob/f8ee506a2a54604b06bc13258f70954fa88f091e/testing/main.cpp.
